.github/workflows/build_docker_image.yml

```diff
@@ -1,12 +1,12 @@
 name: Build Docker Image
 
 on:
+  workflow_dispatch:
   push:
     branches:
-      - main
-  pull_request_target:
-    branches:
-      - main
+      - 'main'
+    tags:
+      - '*.*.*'
```

Comment:
Thanks for the feedback.

In my experience code inspection and build verification are necessary every time CI triggers (if the language needs to be built)

But building the image is unnecessary, because if you don't push it, this process will only cause a waste of time and resources. Regardless of whether insiders are involved.

At present or in the long term, the most frequently updated parts of this warehouse are non-model-related parts, such as documentation and quick start. Perhaps these contents should only be triggered when the maintainer releases a release version?

Want to hear your opinion.

Reply:
> The changes only prevent the build of a docker image upon each pull request. This means that when a pull request is merged, it will still trigger the docker image build.
> 
> Additionally, for pull requests submitted by external contributors, the docker image build process requires explicit authorization from a maintainer. This means it won't trigger automatically for every pull request. You can find the relevant code in https://github.com/01-ai/Yi/blob/main/.github/workflows/build_docker_image.yml#L16-L21

Thanks for asking the question, it's fixed and will only be built when the version is released.

I'd like to submit a new patch in the future about a format review that can be run on every commit.

Just like the points discussed above, in order to make submissions more efficient and high-quality, and reduce unnecessary construction time.

Reply:
> At present or in the long term, the most frequently updated parts of this warehouse are non-model-related parts, such as documentation and quick start. Perhaps these contents should only be triggered when the maintainer releases a release version?

Yes, I agree with you. 

I see a potential benefit in building docker image in pr. This approach allows us to test the docker image created by the specific pull request before it's merged into the main branch. But, as you rightly pointed out, since the most common updates are not code-related, the necessity to update the Docker image for every minor change is indeed debatable. Even when it comes to code changes, we do not need to be so cautious before merging these changes.

@findmyway what do you say?